tagstudio/src/qt/resource_manager.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2024 Travis Abendshien (CyanVoxel).
+# Licensed under the GPL-3.0 License.
+# Created for TagStudio: https://github.com/CyanVoxel/TagStudio
+
+import logging
+from pathlib import Path
+from typing import Any
+
+import ujson
+
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+
+class ResourceManager:
+    """A resource manager for retrieving resources."""
+
+    _map: dict[str, tuple[str, str]] = {}  # dict[<id>, tuple[<filepath>,<mode>]]
+    _cache: dict[str, Any] = {}
+
+    # Initialize _map
+    with open(
+        Path(__file__).parent / "resources.json", mode="r", encoding="utf-8"
+    ) as f:
+        json_map: dict = ujson.load(f)
+        for item in json_map.items():
+            _map[item[0]] = (item[1]["path"], item[1]["mode"])
+
+    logging.info(f"[ResourceManager] Resources Loaded: {_map}")
```

Comment:
- this would be better to do in constructor, otherwise doing `import` of this class will have "side-effect" (ie. opening and loading from json file), similar way as doing `import src.qt.resources_rc` has side-effect of prefilling some QT resources

- when using `.items()`, you can work with it as two variables (key, value), so you dont have to unpack it via numeric index (`item[0]` `item[1]`) later

```diff
-for item in json_map.items():
-    _map[item[0]] = (item[1]["path"], item[1]["mode"])
+ for key, value in json_map.items():
+    _map[key] = (value["path"], value["mode"])
```

- to make this future-proof, I'd assign just the whole `value` instead of extracting specific keys in case there will be some other property in future. This will also mitigate a potential issue further on (will comment below)

```diff
-_map[key] = (value["path"], value["mode"])
+_map[key] = value
```

...and at that point you can do just this:
```diff
-json_map: dict = ujson.load(f)
-for item in json_map.items():
-   _map[item[0]] = (item[1]["path"], item[1]["mode"])
+_map = ujson.load(f)
```